arch/RISCV/riscv_ast.gen.inc

```diff
@@ -0,0 +1,2712 @@
+        RISCV_ANDI
+      } op;
+
+    } itype;
+
+    struct  {
+      uint8_t shamt /* bits : 6 */;
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rd /* bits : 5 */;
+      enum sop {
+        RISCV_SLLI,
+        RISCV_SRLI,
+        RISCV_SRAI
+      } op;
+
+    } shiftiop;
+
+    struct  {
+      uint8_t rs2 /* bits : 5 */;
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rd /* bits : 5 */;
+      enum rop {
+        RISCV_ADD,
+        RISCV_SUB,
+        RISCV_SLL,
+        RISCV_SLT,
+        RISCV_SLTU,
+        RISCV_XOR,
+        RISCV_SRL,
+        RISCV_SRA,
+        RISCV_OR,
+        RISCV_AND
+      } op;
+
+    } rtype;
+
+    struct  {
+      uint16_t imm /* bits : 12 */;
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rd /* bits : 5 */;
+      uint8_t is_unsigned /* bits : 1 */;
+      enum word_width {
+        RISCV_BYTE,
+        RISCV_HALF,
+        RISCV_WORD,
+        RISCV_DOUBLE
+      } width;
+
+      uint8_t aq /* bits : 1 */;
+      uint8_t rl /* bits : 1 */;
+    } load;
+
+    struct  {
+      uint16_t imm /* bits : 12 */;
+      uint8_t rs2 /* bits : 5 */;
+      uint8_t rs1 /* bits : 5 */;
+      enum word_width width;
+      uint8_t aq /* bits : 1 */;
+      uint8_t rl /* bits : 1 */;
+    } store;
+
+    struct  {
+      uint16_t imm /* bits : 12 */;
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rd /* bits : 5 */;
+    } addiw;
+
+    struct  {
+      uint8_t rs2 /* bits : 5 */;
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rd /* bits : 5 */;
+      enum ropw {
+        RISCV_ADDW,
+        RISCV_SUBW,
+        RISCV_SLLW,
+        RISCV_SRLW,
+        RISCV_SRAW
+      } op;
+
+    } rtypew;
+
+    struct  {
+      uint8_t shamt /* bits : 5 */;
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rd /* bits : 5 */;
+      enum sopw {
+        RISCV_SLLIW,
+        RISCV_SRLIW,
+        RISCV_SRAIW
+      } op;
+
+    } shiftiwop;
+
+    struct  {
+      uint8_t pred /* bits : 4 */;
+      uint8_t succ /* bits : 4 */;
+    } fence;
+
+    struct  {
+      uint8_t pred /* bits : 4 */;
+      uint8_t succ /* bits : 4 */;
+    } fence_tso;
+
+    
+    
+    
+    
+    
+    
+    struct  {
+      uint8_t rs1 /* bits : 5 */;
+      uint8_t rs2 /* bits : 5 */;
+    } sfence_vma;
+
+    struct  {
+      uint8_t aq /* bits : 1 */;
+      uint8_t rl /* bits : 1 */;
+      uint8_t rs1 /* bits : 5 */;
+      enum word_width width;
+      uint8_t rd /* bits : 5 */;
+    } loadres;
+
+    struct  {
+      uint8_t aq /* bits : 1 */;
+      uint8_t rl /* bits : 1 */;
+      uint8_t rs2 /* bits : 5 */;
+      uint8_t rs1 /* bits : 5 */;
+      enum word_width width;
+      uint8_t rd /* bits : 5 */;
+    } storecon;
+
+    struct  {
+      enum amoop {
+        RISCV_AMOSWAP,
+        RISCV_AMOADD,
+        RISCV_AMOXOR,
+        RISCV_AMOAND,
+        RISCV_AMOOR,
+        RISCV_AMOMIN,
+        RISCV_AMOMAX,
+        RISCV_AMOMINU,
+        RISCV_AMOMAXU
+      } op;
+
+      uint8_t aq /* bits : 1 */;
```

Comment:
These `/* bits : 1 */` comments, what do they mean?
1. `aq` encodes bit 1 of instruction.
2. `aq` is one bit wide.

Please make this more clear. E.g. for the first meaning you could replace it with `insn_bits[1:1]`. And for the second meaning: `bit_width : 1`.

Reply:
Guess it means the last, though being more descriptive doesn't hurt. This one has a low priority though.